File: Abstraction/Pattern Matching/area.swift
func printArea(_ arg: Any) {
    switch arg {
        case let side as Int:
            print("Area of square is: \(side * side)")
        case let sides as (Int, Int):
            print("Area of rectangle is: \(sides.0 * sides.1)")
        default:
            print("Invalid input")
    }
}

printArea(5)
printArea((4, 6))

File: Abstraction/Pattern Matching/area.kt
fun printArea(arg: Any) {
    when(arg) {
        is Int -> println("Area of square is: ${arg * arg}")
        is Pair<*, *> -> println("Area of rectangle is: ${(arg.first as Int) * (arg.second as Int)}")
    }
}

fun main() {
    printArea(5)
    printArea(Pair(4, 6))
}
